drivers/timer/riscv_machine_timer.c

```diff
@@ -67,7 +67,7 @@ const int32_t z_sys_timer_irq_for_test = TIMER_IRQN;
 
 static uint64_t get_hart_mtimecmp(void)
 {
-	return MTIMECMP_REG + (_current_cpu->id * 8);
+	return MTIMECMP_REG + (csr_read(mhartid) * 8);
```

Comment:
we should start using `arch_proc_id()` instead.

Reply:
Hi @carlocaione, 
will do this now and push shortly.
Cheers!

Reply:
@carlocaione done.